packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -997,11 +1042,15 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		}
 	}
 
-	public getPendingBlobs(): IPendingBlobs {
+	public async getPendingBlobs(waitBlobsToAttach?: boolean): Promise<IPendingBlobs> {
+		if (waitBlobsToAttach) {
+			await this.shutdownPendingBlobs();
+		}
 		const blobs = {};
 		for (const [key, entry] of this.pendingBlobs) {
 			blobs[key] = {
-				blob: bufferToString(entry.blob, "base64"),
```

Comment:
why does this need to change? base64 seems like the correct format for serialization?

Reply:
I have the impression that some conversions were utf8 and others were base64 so it was just a matter of making in consistent. I could change all to base64

Reply:
i'd rather leave it and modify only if it is necessary somewhere like a tests fails. otherwise open an issue to investigate.